### DISCUSSION TITLE
Remove unused CSS variables

### DISCUSSION THREAD

**Discussion by bakura10** [tailwindlabs/tailwindcss#8142] [Ideas]
Hi :),

I am not sure if this is done on purpose but as of today, Tailwind generates a lot of CSS rules with properties like that:

<img width="483" alt="image" src="https://user-images.githubusercontent.com/1198915/163767346-1daf0e46-0a96-4556-859d-f8ed4ec4451b.png">

It is highly improbable for us to use all the properties here, but all those are rendered over and over (this is the same for transform where skew are generated over and over, while we do not use them), giving some clutter to the generated CSS.

I wonder if Tailwind could optimize this by generating the variables dynamically, so if we do not have any class "backdrop-sepia" anywhere, it would not generate the associated CSS variables.

**Comment by maciekpaprocki** [comment]
I agree, with some smaller projects those vars take a lot of space, but also, they are showing with errors, which is pretty annoying. I tried adding prune CSS variables, but it struggles with non root ones. 

Maybe resolving variables at the would be a good idea? the issue with vars is that they are usually longer than actual values so it's not space efficient.
